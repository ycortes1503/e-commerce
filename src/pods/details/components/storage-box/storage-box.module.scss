@use '@/assets/scss/variables/colors' as *;

.container {
  display: flex;

  &__item {
    box-sizing: border-box;
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
    padding: 24px;
    width: 95px;
    height: 65px;
    border: 1px solid $border-grey;
    text-wrap: nowrap;
  }

  &--selected {
    border: 1px solid $black;
  }
}
