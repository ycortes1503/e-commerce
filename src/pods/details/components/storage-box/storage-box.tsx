import * as React from 'react';
import cx from 'classnames';
import * as styles from './storage-box.module.scss';
import { StorageOption } from '@/core';
import { Typography } from '@/common/components';

export interface StorageBoxProps {
  storageOptions: StorageOption[];
  onSelect: (option: StorageOption) => void;
}

export const StorageBox = ({ storageOptions, onSelect }: StorageBoxProps) => {
  const [selected, setSelected] = React.useState<StorageOption>(null);

  const handleSelect = (option: StorageOption) => {
    setSelected(option);
    onSelect(option);
  };

  return (
    <div className={styles.container}>
      {storageOptions.map((option, index) => {
        const key = `${index}-${option}`;

        return (
          <div
            key={key}
            data-testid={`storage-option-${index}`}
            className={cx(styles.container__item, { [styles['container--selected']]: selected === option })}
            onClick={() => handleSelect(option)}
          >
            <Typography fontSize="14" text={option.capacity} />
          </div>
        );
      })}
    </div>
  );
};
