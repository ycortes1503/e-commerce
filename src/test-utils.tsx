import React, { ReactElement } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { CartProvider } from './context';
import { MemoryRouter as Router } from 'react-router-dom';
const WrapperProviders = ({ children }: { children: React.ReactNode }) => {
  return (
    <CartProvider>
      <Router>{children}</Router>
    </CartProvider>
  );
};

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>) =>
  render(ui, { wrapper: WrapperProviders, ...options });

export { customRender as render };

import { MemoryRouter } from 'react-router-dom';

const renderWithRouter = (ui, { route = '/' } = {}) => {
  return render(<MemoryRouter initialEntries={[route]}>{ui}</MemoryRouter>);
};

export { renderWithRouter };

export * from '@testing-library/react';
