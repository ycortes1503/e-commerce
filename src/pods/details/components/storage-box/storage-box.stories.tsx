import { Meta, StoryObj } from '@storybook/react/*';
import { StorageBox } from './storage-box';

const meta = {
  title: 'Molecules/Storage Box',
  component: StorageBox,
} satisfies Meta<typeof StorageBox>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    value: 'Storage value',
  },
};
