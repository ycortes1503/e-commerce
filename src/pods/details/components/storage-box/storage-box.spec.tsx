import { render, screen, fireEvent } from '@/test-utils';
import { StorageBox } from './storage-box';

describe('first', () => {
  const mockFunction = jest.fn();
  const storageOptions = [
    { capacity: '64GB', price: 100 },
    { capacity: '128GB', price: 200 },
  ];

  it('should render correctly', () => {
    const { container } = render(<StorageBox storageOptions={storageOptions} onSelect={mockFunction} />);

    expect(container).toMatchSnapshot();
  });

  it('should call onSelect when a storage option is selected', () => {
    render(<StorageBox storageOptions={storageOptions} onSelect={mockFunction} />);

    const storageBox = screen.getByTestId('storage-option-1');
    fireEvent.click(storageBox);
    expect(mockFunction).toHaveBeenCalledWith(storageOptions[1]);
  });
});
