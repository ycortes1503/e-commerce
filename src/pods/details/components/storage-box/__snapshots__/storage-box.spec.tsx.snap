// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`first should render correctly 1`] = `
<div>
  <div>
    <div
      class=""
      data-testid="storage-option-0"
    >
      <p
        class=""
        style="font-size: 14px;"
      >
        64GB
      </p>
    </div>
    <div
      class=""
      data-testid="storage-option-1"
    >
      <p
        class=""
        style="font-size: 14px;"
      >
        128GB
      </p>
    </div>
  </div>
</div>
`;
